pkg/network/protocols/telemetry/metric.go

```diff
@@ -104,3 +104,35 @@ func (m *metricBase) Get() int64 {
 type metric interface {
 	base() *metricBase
 }
+
+// TLSAwareCounter is a TLS aware counter, it has a plain counter and a counter for TLS.
```

Comment:
moving this struct from http2 to a generic file is great, but it generates a lot of noise in this PR.
we should extract that change into a different PR

Reply:
I removed this struct from metric.go